test/end2end_test.go

```diff
@@ -2432,6 +2432,101 @@ func testHealthCheckOff(t *testing.T, e env) {
 	}
 }
 
+func TestHealthWatch(t *testing.T) {
+	defer leakcheck.Check(t)
+	for _, e := range listTestEnv() {
+		testHealthWatch(t, e)
+	}
+}
+
+func testHealthWatch(t *testing.T, e env) {
+	te := newTest(t, e)
+	hs := health.NewServer()
+
+	service0 := ""
```

Comment:
Could you split this into 2-3 different test functions for the various different things that are being tested by this one?  E.g. `TestHealthWatch_DefaultServiceStartsServing`, `TestHealthWatch_UnknownServiceStartsUnknown`, `TestHealthWatch_MultipleWatchers`?